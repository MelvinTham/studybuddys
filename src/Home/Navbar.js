import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Drawer from '@material-ui/core/Drawer';
import Button from '@material-ui/core/Button';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Menu from '@material-ui/icons/Menu';
import { Link } from 'react-router-dom';

const styles = {
  list: {
    width: 250,
  }
};

class Navbar extends React.Component {
  state = {
    left: false,
  };

  toggleDrawer = (side, open) => () => {
    this.setState({
      [side]: open,
    });
  };

  render() {
    const { classes } = this.props;

    const sideList = (
      <div className={classes.list}>
        <List  style={{width:'1000px'}}>
            <ListItem button >
              <Link 
                to = "/dashboard" 
                style = {{textDecoration: "none", color: "black", width: "100%",height:'50px'}}
              >
                My Courses
              </Link>
            </ListItem >
            <ListItem button > 
              <Link 
                to = "/dashboard/browse" 
                style = {{textDecoration: "none", color: "black", width: '100%',height:'50px'}}
              >
                Course Selection
              </Link>
            </ListItem>
        </List>
      </div>
    );

    return (
      <div>
        <Button style={{float: "left"}} onClick={this.toggleDrawer('left', true)}>
          <Menu fontSize = "large" style = {{color: "white"}}></Menu></Button>
        <Drawer open={this.state.left} onClose={this.toggleDrawer('left', false)}>
          <div
            tabIndex={0}
            role="button"
            onClick={this.toggleDrawer('left', false)}
            onKeyDown={this.toggleDrawer('left', false)}
          >
            {sideList}
          </div>
        </Drawer>
        
      </div>
    );
  }
}

Navbar.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Navbar);