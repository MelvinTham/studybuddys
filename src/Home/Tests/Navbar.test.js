import Enzyme, { render, shallow } from 'enzyme';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import React from 'react';
import Navbar from './../Home';
import { BrowserRouter as Router } from 'react-router-dom';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new enzymeAdapterReact16() });
const getDefaultProps = () => ({});
describe('Navbar component', () => {
  it('shallow renders without crashing', () => {
    const {} = getDefaultProps();
    shallow(
        <Router>
            <Navbar />
        </Router>);
  });
  it('render snapshot', () => {
    const {} = getDefaultProps();
    const wrapper = render(
        <Router>
            <Navbar />
        </Router>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
