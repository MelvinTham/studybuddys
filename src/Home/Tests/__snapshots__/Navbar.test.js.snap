// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component render snapshot 1`] = `
<div
  class="Home"
  data-aos=""
>
  <h1
    class="MuiTypography-root-1 MuiTypography-display4-2"
    data-aos="fade-down"
    data-aos-duration="500"
    data-aos-easing="linear"
    style="color:white;text-shadow:0 0 10px black;margin-bottom:20px"
  >
    Study Buddy's
  </h1>
  <div
    class="MuiPaper-root-37 MuiPaper-elevation8-47 MuiPaper-rounded-38 MuiCard-root-36 signIn"
    data-aos="fade-down"
    data-aos-duration="500"
    data-aos-easing="linear"
  >
    <form>
      <h2
        class="MuiTypography-root-1 MuiTypography-display2-4 MuiTypography-gutterBottom-28"
        style="color:black"
      >
        Sign In
      </h2>
      <div
        class="MuiFormControl-root-64"
      >
        <div
          class="MuiInputBase-root-81 MuiInput-root-68 MuiInput-underline-72 MuiInputBase-formControl-82 MuiInput-formControl-69"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-91 MuiInput-input-76 MuiInputBase-inputType-94 MuiInput-inputType-79"
            placeholder="Enter Your Email"
            type="email"
          />
        </div>
      </div>
      <br />
      <div
        class="MuiFormControl-root-64"
        style="margin-top:15px"
      >
        <div
          class="MuiInputBase-root-81 MuiInput-root-68 MuiInput-underline-72 MuiInputBase-formControl-82 MuiInput-formControl-69"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-91 MuiInput-input-76 MuiInputBase-inputType-94 MuiInput-inputType-79"
            placeholder="Enter Your Password"
            type="password"
          />
        </div>
      </div>
      <br />
      <button
        class="MuiButtonBase-root-124 MuiButton-root-98 MuiButton-outlined-106"
        style="margin-top:15px"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label-99"
        >
          Sign In
        </span>
      </button>
    </form>
    <a
      href="/signup"
      style="margin-top:14px"
      variant="outlined"
    >
      Don't have an account? Sign Up Here
    </a>
  </div>
</div>
`;
